datafusion/physical-plan/src/aggregates/group_values/mod.rs

```diff
@@ -143,8 +148,12 @@ pub fn new_group_values(schema: SchemaRef) -> Result<Box<dyn GroupValues>> {
         }
     }
 
-    if GroupValuesColumn::supported_schema(schema.as_ref()) {
-        Ok(Box::new(GroupValuesColumn::try_new(schema)?))
+    if column::supported_schema(schema.as_ref()) {
```

Comment:
Can you explain here why GroupOrdering::None is required? Is it because the `VectorizedGroupValuesColumn` doesn't keep the groups in order?

If that is the case, it seems like maybe `emit_n` would never be called 🤔

Reply:
> Is it because the VectorizedGroupValuesColumn doesn't keep the groups in order?

Yes, because we iterate all the rows at once, some rows are append beforehand so they are not kept in order.

> If that is the case, it seems like maybe emit_n would never be called

`emit_early_if_necessary` may be called

Reply:
> Can you explain here why GroupOrdering::None is required? Is it because the `VectorizedGroupValuesColumn` doesn't keep the groups in order?
> 
> If that is the case, it seems like maybe `emit_n` would never be called 🤔

The situation is just as @jayzhan211  mentioned, and the detail about why `GroupOrdering::None` needed can also see:
https://github.com/Rachelint/arrow-datafusion/blob/406acb4983efe0c2072c5d7759674eec9db9404a/datafusion/physical-plan/src/aggregates/group_values/column.rs#L792-L834